File: my-ts-react-app/src/app/models/Department.ts
import { Faculty } from "./Faculty";
import { Field } from "./Field";
import { Mentor } from "./Mentor";
import { Student } from "./Student";

export interface Department {
    id: number;
    name: string;
    location: string;
    number: number;
    facultyId: number;
    faculty: Faculty;
    mentors: Mentor[];
    students: Student[];
    fields: Field[];
}

File: Models/Entities/Department.cs
using System.Diagnostics;

namespace DiplomaThesisDigitalization.Models.Entities
{
    public class Department
    {
        public Department()
        {
            this.Mentors = new HashSet<Mentor>();
        }
        public int Id { get; set; }
        public string Name { get; set; }
        public string Location { get; set; }
        public int Number { get; set; }

        public int FacultyId { get; set; }
        public Faculty Faculty { get; set; }
        public virtual ICollection<Mentor> Mentors { get; set; }
        public ICollection<Student> Students { get; set; }
        public ICollection<Field> Fields { get; set; }
    }
}
